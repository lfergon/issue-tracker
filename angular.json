{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "issue-tracker": {
      "projectType": "application",
      "schematics": {},
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/issue-tracker",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": ["zone.js"],
            "tsConfig": "tsconfig.app.json",
            "assets": ["src/favicon.ico", "src/assets"],
            "styles": ["node_modules/@clr/ui/clr-ui.min.css", "src/styles.css"],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "issue-tracker:build:production"
            },
            "development": {
              "buildTarget": "issue-tracker:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "buildTarget": "issue-tracker:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": ["zone.js", "zone.js/testing"],
            "tsConfig": "tsconfig.spec.json",
            "assets": ["src/favicon.ico", "src/assets"],
            "styles": ["src/styles.css"],
            "scripts": []
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": ["src/**/*.ts", "src/**/*.html"]
          }
        }
      }
    }
  },
  "cli": {
    "analytics": "7b562994-7e0c-4e60-a27b-26d736495c7e",
    "schematicCollections": ["@angular-eslint/schematics"]
  }
}
