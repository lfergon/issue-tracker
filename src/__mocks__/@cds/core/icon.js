module.exports = {
  ClarityIcons: {
    addIcons: jest.fn()
  }
};