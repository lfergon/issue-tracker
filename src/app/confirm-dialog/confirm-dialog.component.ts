import {
  Component,
  Output,
  EventEmitter,
  input,
  InputSignal,
  signal,
} from '@angular/core';

@Component({
  selector: 'app-confirm-dialog',
  templateUrl: './confirm-dialog.component.html',
  styleUrls: ['./confirm-dialog.component.css'],
})
export class ConfirmDialogComponent {
  issueNumber = signal(0);
  @Output() confirm = new EventEmitter<boolean>();

  agree() {
    this.confirm.emit(true);
    this.issueNumber = signal(0);
  }

  disagree() {
    this.confirm.emit(false);
    this.issueNumber = signal(0);
  }
}
