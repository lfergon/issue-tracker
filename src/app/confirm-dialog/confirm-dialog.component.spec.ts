import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ConfirmDialogComponent } from './confirm-dialog.component';
import { Component } from '@angular/core';
import { By } from '@angular/platform-browser';
import { ClarityModule } from '@clr/angular';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

window.ResizeObserver =
  window.ResizeObserver ||
  jest.fn().mockImplementation(() => ({
    disconnect: jest.fn(),
    observe: jest.fn(),
    unobserve: jest.fn(),
  }));

@Component({
  template: `
    <app-confirm-dialog
      [issueNumber]="issueNumber"
      (confirm)="onConfirm($event)"
    ></app-confirm-dialog>
  `,
})
class TestHostComponent {
  issueNumber: number | undefined = 123;
  onConfirm(confirm: boolean) {}
}

describe('ConfirmDialogComponent', () => {
  let component: ConfirmDialogComponent;
  let fixture: ComponentFixture<ConfirmDialogComponent>;
  let hostFixture: ComponentFixture<TestHostComponent>;

  beforeEach(async () => {
    window.Element.prototype.animate = jest.fn();

    TestBed.configureTestingModule({
      imports: [ClarityModule, BrowserAnimationsModule],
      declarations: [ConfirmDialogComponent, TestHostComponent],
    });

    fixture = TestBed.createComponent(ConfirmDialogComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    hostFixture = TestBed.createComponent(TestHostComponent);
    component = hostFixture.debugElement.query(
      By.directive(ConfirmDialogComponent),
    ).componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should emit true when agree is called', () => {
    jest.spyOn(component.confirm, 'emit');
    component.agree();
    expect(component.confirm.emit).toHaveBeenCalledWith(true);
  });

  it('should emit false when disagree is called', () => {
    jest.spyOn(component.confirm, 'emit');
    component.disagree();
    expect(component.confirm.emit).toHaveBeenCalledWith(false);
  });

  it('should reset issueNumber when agree is called', () => {
    component.issueNumber = 123;
    component.agree();
    expect(component.issueNumber).toBeUndefined();
  });

  it('should reset issueNumber when disagree is called', () => {
    component.issueNumber = 123;
    component.disagree();
    expect(component.issueNumber).toBeUndefined();
  });

  // TODO: TypeError: Cannot read properties of undefined (reading 'addEventListener')
  // it('should trigger agree method when agree button is clicked', async () => {
  //   jest.spyOn(component, 'agree');
  //   hostFixture.detectChanges();
  //
  //   const button = hostFixture.debugElement.query(By.css('.btn-danger'));
  //   fireEvent.click(button.nativeElement);
  //   expect(component.agree).toHaveBeenCalled();
  // });
  //
  // it('should trigger disagree method when disagree button is clicked', async () => {
  //   jest.spyOn(component, 'disagree');
  //   hostFixture.detectChanges();
  //
  //   const button = hostFixture.debugElement.query(By.css('.btn-outline'));
  //   fireEvent.click(button.nativeElement);
  //   expect(component.disagree).toHaveBeenCalled();
  // });
});
