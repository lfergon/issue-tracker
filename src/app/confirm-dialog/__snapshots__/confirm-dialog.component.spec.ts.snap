// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialogComponent should match snapshot 1`] = `
<app-confirm-dialog
  confirm={[Function EventEmitter_]}
  issueNumber="undefined"
>
  <clr-modal>
    <h3
      class="modal-title"
    >
       Resolve Issue  
    </h3>
    <div
      class="modal-body"
    >
      <p>
        Are you sure you want to close the issue?
      </p>
    </div>
    <div
      class="modal-footer"
    >
      <button
        class="btn btn-outline"
        type="button"
      >
        Cancel
      </button>
      <button
        class="btn btn-danger"
        type="button"
      >
        Yes, continue
      </button>
    </div>
  </clr-modal>
</app-confirm-dialog>
`;
